### DISCUSSION TITLE
custom widgets, display external images by given url

### DISCUSSION THREAD

**Discussion by rademakeronline** [Zefau/ioBroker.jarvis#605] [Q&A] [answered]
Dear community,

I wonder if there is a way to display an external image by a given URL in the statelist by "states-icon-display" or "states-icon-stateStyle" ?

Another question is, if it's possible to change the whole span tag of a data point to an img or div tag?

And finally: Is there a way to define custom widgets?

**Comment by Zefau** [comment, marked as answer]
Have you had a look into the [FAQ](https://github.com/Zefau/ioBroker.jarvis/wiki/de-FAQ) ?

1. You may convert any image to base64 and use it as an icon within the Device. This is will be used in StateList. An URL is not possible.

2. Why would you want to change `span` to `img` / `div` ? What do you want to archieve? You may display an image by using HTML code (`<img ... />`) within a state (in ioBroker) and then using that state in jarvis. StateList will display HTML code and thus show the image.

3. What do you exactly mean by Custom Widgets? What do you wanna display? You may create a custom widget in ioBroker.vis and display that via iFrame.

**Reply by rademakeronline** [reply]
Dear Zefau,

thanks for the response. In general, I want to display images dynamically in the StateList. I think with the solution mentioned in 2, I will achieve my needs. 

With Custom Widgets mean, define widgets with custom layouts and or grids. But if there is no such option, it doesn't matter. I just couldn't find anything about it in the documentation and thought it would be a cool feature.

Greetings from Lower Saxony 
Christian

**Reply by Zefau** [reply]
> With Custom Widgets mean, define widgets with custom layouts and or grids. But if there is no such option, it doesn't matter. I just couldn't find anything about it in the documentation and thought it would be a cool feature.

You may define any number of columns and any number of widgets within these columns. With future versions widgets more span across columns and also have different sizes.
